### PR TITLE
[TextParagraph/Button] Add support for line spacing.

### PR DESCRIPTION
**Pull request by bruvzg**
Fixes https://github.com/godotengine/godot-proposals/discussions/10476

Related proposal https://github.com/godotengine/godot-proposals/issues/9778 (adds support to `TextParagraph` but not to all controls using it).

### CODE REVIEW

File: doc/classes/Button.xml [hunk 1]
```diff
@@ -127,6 +127,9 @@
 		<theme_item name="icon_max_width" data_type="constant" type="int" default="0">
 			The maximum allowed width of the [Button]'s icon. This limit is applied on top of the default size of the icon, or its expanded size if [member expand_icon] is [code]true[/code]. The height is adjusted according to the icon's ratio. If the button has additional icons (e.g. [CheckBox]), they will also be limited.
 		</theme_item>
+		<theme_item name="line_spacing" data_type="constant" type="int" default="0">
+			Vertical spaceing between lines.
```
Comment (Mickeon) [inline doc/classes/Button.xml:131]:
```suggestion
			Vertical spacing between lines.
```

File: doc/classes/TextParagraph.xml [hunk 2]
```diff
@@ -280,6 +280,9 @@
 		<member name="justification_flags" type="int" setter="set_justification_flags" getter="get_justification_flags" enum="TextServer.JustificationFlag" is_bitfield="true" default="163">
 			Line fill alignment rules. See [enum TextServer.JustificationFlag] for more information.
 		</member>
+		<member name="line_spacing" type="float" setter="set_line_spacing" getter="get_line_spacing" default="0.0">
+			Vertical spaceing between lines.
```
Comment (Mickeon) [inline doc/classes/TextParagraph.xml:284]:
```suggestion
			Vertical spacing between lines.
```

**Comment by timothyqiu** [comment]
What's the unit of line spacing? Also, I think we should make it clear in the documentation whether "line spacing" here means:

- The distance between adjacent baselines, or
- The distance between the bottom of one line and the top of the next.

**Comment by bruvzg** [comment]
> What's the unit of line spacing?

Pixel.

> The distance between adjacent baselines, or
> The distance between the bottom of one line and the top of the next.

It's additional spacing, not a total spacing, so in does not matter from what point it's measured.

It's used in the same way existing `line_spacing` in `Label`/`Label3D`/`TextMesh` and `line_separation` in `RichTextLabel` work, spacing is added to the line descent:
```
    pos.y += line_ascent;
    draw_line(pod, ...); // draw use baseline
    pos.y += line_descent + line_spacing; 
```

**Review by Calinou — APPROVED** [review]
Code looks good to me. It makes sense to expose this in Button since multiline buttons are fully supported in 4.x.

**Review by KoBeWi — APPROVED** [review]

**Review by AThousandShips — APPROVED** [review]

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**